File: Act11/Act11.py
def saludar():
    print("Hola mundo!")

def calcular_doble(num):
    return num * 2

def triplicar(num_ref):
    num_ref[0] = num_ref[0] * 3  

def prueba_funciones():
    print("Llamada a la funcion Saludar:")
    saludar()

    x = int(input("Ingrese un valor numerico para x: "))

    print("Llamada a la funcion CalcularDoble (pasaje por valor)")
    print(f"El doble de {x} es {calcular_doble(x)}")
    print(f"El valor original de x es {x}")

    print("Llamada a la funcion Triplicar (pasaje por referencia)")
    x_ref = [x]  
    triplicar(x_ref)
    print(f"El nuevo valor de x es {x_ref[0]}")

prueba_funciones()

File: Act11/Act11.cpp
#include <iostream>

using namespace std;

void Saludar() {
    cout << "Hola mundo!" << endl;
}

int CalcularDoble(int num) {
    return num * 2;
}

void Triplicar(int &num) {
    num = num * 3;
}

int main() {
    cout << "Llamada a la funcion Saludar:" << endl;
    Saludar();

    int x;
    cout << "Ingrese un valor numerico para x: ";
    cin >> x;

    cout << "Llamada a la función CalcularDoble (pasaje por valor)" << endl;
    cout << "El doble de " << x << " es " << CalcularDoble(x) << endl;
    cout << "El valor original de x es " << x << endl;

    cout << "Llamada a la funcion Triplicar (pasaje por referencia)" << endl;
    Triplicar(x);
    cout << "El nuevo valor de x es " << x << endl;

    return 0;
}
